composables/auto-prompter.ts

```diff
@@ -294,25 +301,193 @@ export function useAutoPrompter() {
 
         newBattle.winner = winner
         battles.value.push(newBattle)
+    }
 
-        takeSnapshotOfRatings()
+    // define ammount of battles and prompts in each round
+    function defineBattlesPerRound (numberOfPrompts: number, numberOfBattles: number) {
+        const promptsPerRound: number[] = []
+        const battlesPerRound: number[] = []
+        let sumOfPromptsPerRound = 0
+        let sumOfBattlesPerRound = 0
+
+        // calculate number of prompts that will be alive in each round
+        promptsPerRound[0] = numberOfPrompts
+        for (let i = 1; i < Math.floor(Math.log2(numberOfPrompts)); i++) {
+            promptsPerRound[i] = promptsPerRound[i-1] - Math.ceil(promptsPerRound[i-1]/2)
+        }
+
+        //calculate ammount of prompts-round for wheighing battle distribution
+        for (const p of promptsPerRound){
+            sumOfPromptsPerRound += p
+        }
+        
+        for (let i = 0; i < promptsPerRound.length; i++) {
+        battlesPerRound [i] = Math.floor(promptsPerRound[i]/sumOfPromptsPerRound*numberOfBattles)
+        }
+
+        for (const b of battlesPerRound){
+            sumOfBattlesPerRound += b
+        }
+
+        let remainder = numberOfBattles - sumOfBattlesPerRound
+
+        for (let i = 0; i < remainder; i++){
+            battlesPerRound[i] += 1
+        }
+
+        return battlesPerRound
+    }
+
+    async function generateCandidateVariations (candidate : Candidate){
+        const ai = useAI()
+        const {
+            candidateGenerationModel,
+            candidateGenerationTemperature,
+        } = useSettings()
+        const newPrompt = `You are provided with :
```

Comment:
```suggestion
        const newPrompt = trim`You are provided with :
```

Use the `trim` util here to automatically trim unwanted leading whitespace.